Cookie poisoning (#114)
ShiftLeftSecurity/documentation: using-ocular/use-cases/cookie-poisoning.md

> # HTTP Cookie Poisoning

Reviewer: Mind renaming the file too? eg `http-cookie-poisoning.md`

Author: Done

> The main purpose of a cookie is to identify users and possibly prepare customized Web pages for consumers based on their profile. There are three areas of potential security risks with cookies: 

Reviewer: We should include a short definition of cookie poisoning: the act of an attacker modifying the cookie for the purpose of abusing the functionality of an application.

Author: Added

> The main purpose of a cookie is to identify users and possibly prepare customized Web pages for consumers based on their profile. There are three areas of potential security risks with cookies: 

Author: Done.

> * Sensitive information may be added by the cookie as plain text, which is not recommended. A malicious user can sniff the connection to capture any data that is being transmitted over a network. Therefore, some sort of encryption should be used.

Reviewer: ```suggestion
* Sensitive information may be added by the cookie as plain text, which is not recommended. A malicious user can sniff the connection to capture any data that is being transmitted over a network. Therefore, encryption should be used.
```

I'd remove "some sort of", because there's a clear standard about what kind of encryption is most commonly used in cookies.

Author: Done.

> * The developer may not have set the HttpOnly attribute. HttpOnly is an instruction for the browser to not change the cookie nor give it to javascript. Setting the HttpOnly attribute protects from an XSS vulnerability.

Reviewer: ```suggestion
* The developer may not have set the `HttpOnly` attribute. `HttpOnly` is an instruction for the browser to not change the cookie nor give it to the JavaScript engine. Setting the `HttpOnly` attribute limits the damage done in case of an XSS vulnerability.
```

Author: Done

> * The developer may not have set the HttpOnly attribute. If the browser supports the HttpOnly attribute (as in fact most  browsers do), the browser does not allow the value of the cookie to change. However, for example if the user type is added to the cookie, and the user changes the user type from User to Admin, then the application accepts the next requests of this user as an admin request.

Reviewer: This is not a great example. HttpOnly is an instruction for the browser to not change the cookie nor give it to javascript. Although an attacker can just skip using a browser altogether and still swap out the cookie.

The best thing `HttpOnly` protects from is in case of an XSS vulnerability, an attacker would not be able to steal another user's cookies, because the javascript in the browser cannot see them.

Author: Rewrote

> * The developer did not set the Secure attribute. Developers usually store all sorts of information in cookies. Because cookies are easy to manipulate, they can exchange information from the server to the browser and vice-versa. 

Reviewer: We should clarify that the risk is a man-in-the-middle attack when `Secure` attribute is not used.

Author: Rewritten

> * The developer did not set the Secure attribute. Developers usually store all sorts of information in cookies. Because cookies are easy to manipulate, they can exchange information from the server to the browser and vice-versa. When the Secure attribute not used, the risk is a man-in-the-middle attack.

Reviewer: ```suggestion
* The developer did not set the `Secure` attribute. Developers usually store all sorts of information in cookies. Because cookies are easy to manipulate, they can exchange information from the server to the browser and vice-versa. When the `Secure` attribute is not used, the risk is a man-in-the-middle attack.
```

Author: Done
